cmd/kubelet/app/options/options.go

```diff
@@ -216,4 +215,5 @@ func (s *KubeletServer) AddFlags(fs *pflag.FlagSet) {
 	fs.DurationVar(&s.OutOfDiskTransitionFrequency.Duration, "outofdisk-transition-frequency", s.OutOfDiskTransitionFrequency.Duration, "Duration for which the kubelet has to wait before transitioning out of out-of-disk node condition status. Default: 5m0s")
 	fs.StringVar(&s.NodeIP, "node-ip", s.NodeIP, "IP address of the node. If set, kubelet will use this IP address for the node")
 	fs.BoolVar(&s.EnableCustomMetrics, "enable-custom-metrics", s.EnableCustomMetrics, "Support for gathering custom metrics.")
+	fs.StringVar(&s.RuntimeContainer, "runtime-container", s.RuntimeContainer, "Absolute name of the cgroups to create (if required) and run the runtime in (Default: /docker-daemon).")
```

Comment:
This PR ensures that we will discover the docker daemon's container, if this flag is set to `""`. So thats taken care of.
The only question that still left is that of defaults. There are at-least two sets of users - systemd, non-systemd (maybe more). Each party wants to setup the node differently. We can either recognize each type of configuration as a first-class entity in kubelet, or let users express their node configuration explicitly.
If we go with the former, we will have a built-in policy for systemd, one for GCP, and possible more.
With this model, we will publish a cookbook and avoid needing inbuilt models. WDYT?